src/Microsoft.Restier.Core/Submit/ChangeSetItem.cs

```diff
@@ -148,16 +151,18 @@ public class DataModificationItem : ChangeSetItem
         /// </param>
         public DataModificationItem(
             string entitySetName,
-            string entityTypeName,
+            Type entityType,
```

Comment:
how about `expectedEntityType` (the expected entity type from the model)  and `actualEntityType`?